src/tests.rs

```diff
@@ -1,99 +1,130 @@
 #[cfg(test)]
 mod unit {
-    fn check(data: &[u8]) -> Result<Vec<crate::Item>, std::io::Error> {
+    use crate::Error;
+
+    fn check_slice(mut data: &[u8]) -> Result<Vec<crate::Item>, Error> {
+        let mut items = vec![];
+        while let Some((item, rest)) = crate::read_one_from_slice(data)? {
+            items.push(item);
+            data = rest;
+        }
+
+        Ok(items)
+    }
+
+    fn check_io(data: &[u8]) -> Result<Vec<crate::Item>, std::io::Error> {
```

Comment:
I think it might make sense to import`Item` like `Error` to avoid needing the `crate::` qualifier throughout.